.circleci/config.yml

```diff
@@ -46,7 +46,7 @@ jobs:
             docker run --rm \
               -v $PWD:/code \
               -w /code \
-              rust:1.72.0-alpine3.18 \
```

Comment:
The 1.73 image is not ready yet, but I wanted to adjust this 1 of 3 image names used in CI that was out of sync.

Reply:
FWIW, to sanity check all this and see a new 1.73.0 feature I did:
```
$ docker run --rm -v $PWD:/code -w /code rust:1.72.1-alpine3.18 sh -c 'apk add cmake make musl-dev perl && cargo run -p package -- dist'
Unable to find image 'rust:1.72.1-alpine3.18' locally
1.72.1-alpine3.18: Pulling from library/rust
...
  Installing /code/target/debug/build/package-de1c79de9adcfca8/out/dist/bin/ptex
   Installed package `ptex v0.7.0 (/code)` (executable `ptex`)
Wrote the ptex (target: x86_64-unknown-linux-musl) to dist/ptex-linux-x86_64
$ readelf -p.comment target/debug/build/package-de1c79de9adcfca8/out/dist/bin/ptex

String dump of section '.comment':
  [     0]  GCC: (GNU) 9.4.0
  [    11]  rustc version 1.73.0 (cc66ad468 2023-10-03)
  [    3d]  GCC: (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924
```

And that's vs older Rust:
```
$ readelf -p.comment `which science`

String dump of section '.comment':
  [     0]  GCC: (GNU) 9.4.0
  [    11]  GCC: (Alpine 12.2.1_git20220924-r10) 12.2.1 20220924
```